modmain.lua

```diff
@@ -453,9 +461,14 @@ local function StatusPostConstruct(self)
 	local nudge = 0
 	if SHOWNAUGHTINESS then	
 		self.naughtiness = self:AddChild(Minibadge("naughtiness", self.owner))
-		local function UpdateNaughty()
-			self.naughtiness.num:SetString(	(self.owner.components.kramped.actions or 0) .. "/" ..
-											(self.owner.components.kramped.threshold or 0) 			)
+		local function UpdateNaughty(_, data) -- player, data
+			if DST == false then
+				self.naughtiness.num:SetString(	(self.owner.components.kramped.actions or 0) .. "/" ..
+												(self.owner.components.kramped.threshold or 0) 			)
+			else
+				-- data = { actions = n, threshold = n }
+				self.naughtiness.num:SetString(data.actions .. "/" .. data.threshold)
+			end

```

Comment:
I'd suggest adding a guard on this, mainly to defend against other mods potentially pushing this event:

```suggestion
			if DST then
				data = type(data) == "table" and data or {}
			else
				data = self.owner.components.kramped
			end
			local actions = type(data.actions) == "number" and data.actions or 0
			local threshold = type(data.threshold) == "number" and data.threshold or 0
			self.naughtiness.num:SetString(actions .. "/" .. threshold)
```